codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/customization/presignable/PresignableUrlIntegration.kt

```diff
@@ -86,6 +89,9 @@ class PresignableUrlIntegration(private val presignedOperations: Map<String, Set
                 "com.amazonaws.s3#PutObject" -> {
                     renderMiddlewareClassForPutObject(ctx, delegator, op)
                 }
+                "com.amazonaws.s3#UploadPart" -> {
+                    renderMiddlewareClassForUploadPart(ctx, delegator, op)
+                }
```

Comment:
As per previous comment, to use the already generated `queryItemProvider` for `UploadPartInput`, you won't need to add this custom middleware here anymore.